docs/visual-basic/programming-guide/language-features/arrays/index.md

```diff
@@ -14,257 +12,311 @@ helpviewer_keywords:
- The following example has an array of months, each element of which is an array of days. Because different months have different numbers of days, the elements don’t form a rectangular two-dimensional array. Therefore, a jagged array is used instead of a multidimensional array.  
-  
- [!code-vb[VbVbalrArrays#21](../../../../../samples/snippets/visualbasic/VS_Snippets_VBCSharp/VbVbalrArrays/VB/Class1.vb#21)]  
-  
-##  <a name="BKMK_ZeroLength"></a> Zero-Length Arrays  
- An array that contains no elements is also called a zero-length array. A variable that holds a zero-length array doesn’t have the value `Nothing`. To create an array that has no elements, declare one of the array's dimensions to be -1, as the following example shows.  
-  
- [!code-vb[VbVbalrArrays#14](../../../../../samples/snippets/visualbasic/VS_Snippets_VBCSharp/VbVbalrArrays/VB/Class1.vb#14)]  
-  
- You might need to create a zero-length array under the following circumstances:  
+ [!code-vb[iterate-one-dimensional-array](../../../../../samples/snippets/visualbasic/programming-guide/language-features/arrays/iterate1d.vb)]  
   
--   Without risking a <xref:System.NullReferenceException> exception, your code must access members of the <xref:System.Array> class, such as <xref:System.Array.Length%2A> or <xref:System.Array.Rank%2A>, or call a [!INCLUDE[vbprvb](~/includes/vbprvb-md.md)] function such as <xref:Microsoft.VisualBasic.Information.UBound%2A>.  
+ The following example iterates through a multidimensional array by using a [`For...Next`](../../../../visual-basic/language-reference/statements/for-next-statement.md) statement. The <xref:System.Array.GetUpperBound%2A> method has a parameter that specifies the dimension. `GetUpperBound(0)` returns the highest index of the first dimension, and `GetUpperBound(1)` returns the highest index of the second dimension.
   
--   You want to keep the consuming code simpler by not having to check for `Nothing` as a special case.  
+ [!code-vb[iterate-two-dimensional-array](../../../../../samples/snippets/visualbasic/programming-guide/language-features/arrays/iterate2d.vb)]  
   
--   Your code interacts with an application programming interface (API) that either requires you to pass a zero-length array to one or more procedures or returns a zero-length array from one or more procedures.  
+ The following example uses a [For Each...Next Statement](../../../../visual-basic/language-reference/statements/for-each-next-statement.md)to iterate through a one-dimensional array and a two-dimensional array.  
   
-##  <a name="BKMK_ArraySize"></a> Array Size  
- The size of an array is the product of the lengths of all its dimensions. It represents the total number of elements currently contained in the array.  
+ [!code-vb[iterate-for-each-next](../../../../../samples/snippets/visualbasic/programming-guide/language-features/arrays/iterate-for-each-next.vb)]  
   
- The following example declares a three-dimensional array.  
+## Array Size  
```

Comment:
I would mention somewhere in this section that none of this applies to jagged arrays (also referred to as "arrays of arrays")

Reply:
@BillWagner, yes. Your comment points to a weakness of the existing topic -- I didn't discuss determining the size of a jagged array, so I'll add that and an example to the jagged arrays section. When I do that (hopefully later today, but maybe not until tomorrow) I'd like you to take a look at the latest round of revisions.